miprometheus/problems/question_context_to_class/babiqa_dataset_single_question.py

```diff
@@ -0,0 +1,514 @@
+
+        return loss
+
+
+    def build_dictionaries_one_hot(self):
+
+        """Creates the word embeddings for BABI QA with one hot vectors
+
+                - 1. Collects all datasets word
+                - 2. Uses Language object to create the embeddings
+
+                 If it is the first time you run this code, it will take longer to load the embedding from torchtext
+                 """
+        #load data
+        data = self.load_data(tasks=self.tasks, tenK=self.tenK, add_punctuation=True, data_type='train', outmod="one_hot")
+        data = data + self.load_data(tasks=self.tasks, tenK=self.tenK, add_punctuation=True, data_type='valid',
+                                     outmod="one_hot")
+        data = data + self.load_data(tasks=self.tasks, tenK=self.tenK, add_punctuation=True, data_type='test',
+                                     outmod="one_hot")
+
+        # make placeholders dictionnaries with special caracters
+        answ_to_ix = {".": 0, "?": 1, "_": 2}
+        itos_d = [".", "?", "_"]
+
+        # display a progress bar while going through the data
+        self.fix_length = 0
+        for q in tqdm(data):
+            story, answers = q
+            self.fix_length = max(self.fix_length, len(story))
+
+            # go through all the stories
+            for answer in story:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a,ix)
+
+            #go through all the answers
+            for answer in answers:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a, ix)
+
+        #return the corresponding dictionnaries
+        ret = (answ_to_ix)
+        return ret, itos_d
+
+    def build_dictionaries(self):
+
+        """Creates the word embeddings BABI QA 
+
+        - 1. Collects all datasets word
+        - 2. Uses Language object to create the embeddings
+
+         If it is the first time you run this code, it will take longer to load the embedding from torchtext
+         """
+
+        print(' ---> Constructing the dictionaries with word embedding, may take some time ')
+
+        # making an empty list of words meant to store all possible datasets words
+        text = []
+        tasks = self.tasks
+
+        data = self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='train', outmod="embedding")
+        data = data + self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='valid',
+                                     outmod="embedding")
+        data = data + self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='test',
+                                     outmod="embedding")
+
```

Comment:
What is going in here?? It seems that we are merging training with validation with test tests together?

Is that ok?